### PR TITLE
Homework 9 task 2

### PR DESCRIPTION
**Pull request by ivanzuev78**

### CODE REVIEW

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
> Do it both ways: as a class and as a generator.

Does both solution have the same behavior?

**Comment by ivanzuev78** [comment]
I suppose they have a little bit different behaviour, but they have the same result. 
I don't know how we catch an exception in the function context manager.

**Comment by Cjkjvfnby** [comment]
> but they have the same result

Could you please add 2 tests

```python
def test_....():
    with XXX(Exception):
        raise IndexError("Raise IndexError from SuppressorClass")
```

File: HomeWork_9/test_task_2.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from HomeWork_9.hw2 import SuppressorClass, suppressor_function
+
+import pytest
+
+
+def test_suppressor_class_catch_error():
+    with SuppressorClass(IndexError):
+        raise IndexError("Raise IndexError from SuppressorClass")
+    assert True
```
Comment (Cjkjvfnby) [inline HomeWork_9/test_task_2.py:9]: Why do you need it?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_9/test_task_2.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_9/test_task_2.py:9]: Just for beauty.
We don't need it at all.

> > but they have the same result
> 
> Could you please add 2 tests
> 
> ```python
> def test_....():
>     with XXX(Exception):
>         raise IndexError("Raise IndexError from SuppressorClass")
> ```

Hmmm, SuppressorClass doesn't work, but suppressor_function works. Interesting.

Comment (ivanzuev78) [reply HomeWork_9/test_task_2.py:9]: I understand now. I hadn't thought about inheritance. 
Now it should work correctly.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**